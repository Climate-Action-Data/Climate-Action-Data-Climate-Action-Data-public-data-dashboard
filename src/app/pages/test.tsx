'use client'
import { type NextPage } from "next";

const Test: NextPage = () => {
    return (
        <div>
            <h1>Hello</h1>
            <p>data</p>
        </div>
    )
}

export default Test