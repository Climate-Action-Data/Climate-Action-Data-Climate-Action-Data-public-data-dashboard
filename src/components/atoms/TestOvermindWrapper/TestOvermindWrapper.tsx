import { createOvermind } from 'overmind'
import { Provider } from 'overmind-react'
import { config } from '@/overmind'
import { PropsWithChildren } from 'react'

const overmind = createOvermind(config)

export const TestOvermindWrapper = (props: PropsWithChildren) => {
  return <Provider value={overmind}>{props.children}</Provider>
}
