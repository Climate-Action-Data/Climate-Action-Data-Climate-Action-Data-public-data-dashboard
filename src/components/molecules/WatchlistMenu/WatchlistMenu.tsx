import { Menu, MenuButton, MenuItem, Text, IconButton, MenuList } from '@chakra-ui/react'
import { useTranslation } from 'react-i18next'
import { ThreeDotsIcon } from '@/components/atoms/ThreeDotsIcon/ThreeDotsIcon'
import { TrashIcon } from '@/components/atoms/TrashIcon/TrashIcon'
import { DeletePopup } from '@/components/atoms/DeletePopup/DeletePopup'
import { useState } from 'react'
import { useEffects } from '@/overmind'
import { useRouter } from 'next/navigation'
import { EditWatchlistPopup } from '@/components/atoms/EditWatchlistPopup/EditWatchlistPopup'
import { Watchlist } from '@/@types/Watchlist'

interface WatchlistMenuProps {
  watchlist: Watchlist
}

export const WatchlistMenu = (props: WatchlistMenuProps) => {
  const { watchlist } = props
  const { t } = useTranslation(`watchlist`)
  const router = useRouter()

  const [showDeletePopup, setShowDeletePopup] = useState<boolean>(false)
  const [showRenamePopup, setShowRenamePopup] = useState<boolean>(false)

  const { renameWatchlist, deleteWatchlist } = useEffects().watchlist

  const handleRename = () => {
    setShowRenamePopup(true)
  }

  const handleRenameConfirm = async (name: string, description: string) => {
    await renameWatchlist(watchlist.id, name, description)
    setShowRenamePopup(false)
    router.push(`/watchlist/view?id=${watchlist.id}`)
  }

  const handleRenameCancel = () => {
    setShowRenamePopup(false)
  }

  const handleDelete = () => {
    setShowDeletePopup(true)
  }

  const handleDeleteConfirm = () => {
    deleteWatchlist(watchlist.id)
    setShowDeletePopup(false)
    router.push(`/watchlist/all`)
  }

  const handleDeleteCancel = () => {
    setShowDeletePopup(false)
  }

  return (
    <>
      <Menu variant="menuWhite">
        <MenuButton alignSelf="end" as={IconButton} textAlign="center" icon={<ThreeDotsIcon w="32px" height="32px" />} variant="lightGrayRound32"></MenuButton>
        <MenuList>
          <MenuItem data-testid="watchlist-rename" onClick={handleRename} minH="48px">
            <Text flex={1} as="span">
              {t(`renameWatchlist`)}
            </Text>
          </MenuItem>
          <MenuItem data-testid="watchlist-delete" onClick={handleDelete} minH="48px">
            <Text color="red.400" flex={1} as="span">
              {t(`deleteWatchlist`)}
            </Text>
            <TrashIcon color={`red.400`} />
          </MenuItem>
        </MenuList>
      </Menu>
      <DeletePopup
        isOpen={showDeletePopup}
        title={t(`deleteWatchlistTitle`)}
        description={t(`deleteWatchlistText`)}
        onCancel={handleDeleteCancel}
        onConfirm={handleDeleteConfirm}
      />
      <EditWatchlistPopup
        isOpen={showRenamePopup}
        title={t(`renameWatchlist`)}
        name={watchlist.name}
        description={watchlist.description}
        onCancel={handleRenameCancel}
        onConfirm={handleRenameConfirm}
      />
    </>
  )
}
