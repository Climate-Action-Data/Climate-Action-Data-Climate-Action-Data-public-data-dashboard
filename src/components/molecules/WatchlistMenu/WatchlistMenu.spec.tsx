import { render, screen, fireEvent } from '@testing-library/react'
import { WatchlistMenu } from './WatchlistMenu'
import { TestRouter } from '@/test/TestRouter'
import { TestOvermindWrapper } from '@/test/TestOvermindWrapper'
import { watchlistData } from '@/test/mock-data/watchlist_data'

it(`renders correctly`, () => {
  const { container } = render(
    <TestOvermindWrapper>
      <TestRouter router={{}}>
        <WatchlistMenu watchlist={watchlistData[0]} />
      </TestRouter>
    </TestOvermindWrapper>,
  )
  expect(container).toMatchSnapshot()
})

it(`renders correctly and click rename`, () => {
  const { container } = render(
    <TestOvermindWrapper>
      <TestRouter router={{}}>
        <WatchlistMenu watchlist={watchlistData[0]} />
      </TestRouter>
    </TestOvermindWrapper>,
  )
  const renameButton = screen.getByTestId(`watchlist-rename`)
  fireEvent.click(renameButton)
  expect(container).toMatchSnapshot()
})

const DEFAULT_RENDER_TIMEOUT = 2000

it(`renders correctly and click delete`, async () => {
  const mockPush = jest.fn()
  const { container } = render(
    <TestOvermindWrapper>
      <TestRouter router={{ push: mockPush }}>
        <WatchlistMenu watchlist={watchlistData[0]} />
      </TestRouter>
    </TestOvermindWrapper>,
  )
  const renameButton = screen.getByTestId(`watchlist-delete`)
  fireEvent.click(renameButton)
  await new Promise((r) => setTimeout(r, DEFAULT_RENDER_TIMEOUT))
  const confirmModal = screen.getByTestId(`confirm-modal`)
  fireEvent.click(confirmModal)
  expect(mockPush).toHaveBeenCalled()
  expect(container).toMatchSnapshot()
})
