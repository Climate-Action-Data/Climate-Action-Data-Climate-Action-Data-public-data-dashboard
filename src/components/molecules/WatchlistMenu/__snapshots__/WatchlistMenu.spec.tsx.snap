// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    aria-controls="menu-list-:r1:"
    aria-expanded="false"
    aria-haspopup="menu"
    class="chakra-button chakra-menu__menu-button css-1htmeuh"
    id="menu-button-:r1:"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-38f4wb"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M4.73118 12C4.73118 11.5875 4.87805 11.2344 5.17178 10.9406C5.46553 10.6469 5.81865 10.5 6.23113 10.5C6.64363 10.5 6.99675 10.6469 7.29048 10.9406C7.58423 11.2344 7.73111 11.5875 7.73111 12C7.73111 12.4125 7.58423 12.7656 7.29048 13.0593C6.99675 13.3531 6.64363 13.4999 6.23113 13.4999C5.81865 13.4999 5.46553 13.3531 5.17178 13.0593C4.87805 12.7656 4.73118 12.4125 4.73118 12ZM10.5004 12C10.5004 11.5875 10.6473 11.2344 10.941 10.9406C11.2348 10.6469 11.5879 10.5 12.0004 10.5C12.4128 10.5 12.766 10.6469 13.0597 10.9406C13.3535 11.2344 13.5003 11.5875 13.5003 12C13.5003 12.4125 13.3535 12.7656 13.0597 13.0593C12.766 13.3531 12.4128 13.5 12.0004 13.5C11.5879 13.5 11.2348 13.3531 10.941 13.0593C10.6473 12.7656 10.5004 12.4125 10.5004 12ZM16.2696 12C16.2696 11.5875 16.4165 11.2344 16.7102 10.9406C17.004 10.6469 17.3571 10.5 17.7696 10.5C18.1821 10.5 18.5352 10.6469 18.8289 10.9406C19.1227 11.2344 19.2695 11.5875 19.2695 12C19.2695 12.4125 19.1227 12.7656 18.8289 13.0593C18.5352 13.3531 18.1821 13.5 17.7696 13.5C17.3571 13.5 17.004 13.3531 16.7102 13.0593C16.4165 12.7656 16.2696 12.4125 16.2696 12Z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-sjt5nk"
      id="menu-list-:r1:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        class="chakra-menu__menuitem css-1eocwdv"
        data-index="0"
        data-testid="watchlist-rename"
        id="menu-list-:r1:-menuitem-:r2:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          class="chakra-text css-1rr4qq7"
        >
          Rename watchlist
        </span>
      </button>
      <button
        class="chakra-menu__menuitem css-1eocwdv"
        data-index="1"
        data-testid="watchlist-delete"
        id="menu-list-:r1:-menuitem-:r3:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          class="chakra-text css-rzksxu"
        >
          Delete watchlist
        </span>
        <svg
          class="chakra-icon css-1tqudqa"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.3077 20.4998C6.80898 20.4998 6.38302 20.3232 6.02982 19.97C5.67661 19.6168 5.5 19.1908 5.5 18.6921V5.99981H4.5V4.49983H8.99997V3.61523H15V4.49983H19.5V5.99981H18.5V18.6921C18.5 19.1972 18.325 19.6248 17.975 19.9748C17.625 20.3248 17.1974 20.4998 16.6922 20.4998H7.3077ZM17 5.99981H6.99997V18.6921C6.99997 18.7818 7.02883 18.8556 7.08652 18.9133C7.14422 18.971 7.21795 18.9998 7.3077 18.9998H16.6922C16.7692 18.9998 16.8397 18.9678 16.9038 18.9037C16.9679 18.8395 17 18.769 17 18.6921V5.99981ZM9.40385 16.9998H10.9038V7.99981H9.40385V16.9998ZM13.0961 16.9998H14.5961V7.99981H13.0961V16.9998Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly and click rename 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
>
  <button
    aria-controls="menu-list-:r7:"
    aria-expanded="false"
    aria-haspopup="menu"
    class="chakra-button chakra-menu__menu-button css-1htmeuh"
    id="menu-button-:r7:"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-38f4wb"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M4.73118 12C4.73118 11.5875 4.87805 11.2344 5.17178 10.9406C5.46553 10.6469 5.81865 10.5 6.23113 10.5C6.64363 10.5 6.99675 10.6469 7.29048 10.9406C7.58423 11.2344 7.73111 11.5875 7.73111 12C7.73111 12.4125 7.58423 12.7656 7.29048 13.0593C6.99675 13.3531 6.64363 13.4999 6.23113 13.4999C5.81865 13.4999 5.46553 13.3531 5.17178 13.0593C4.87805 12.7656 4.73118 12.4125 4.73118 12ZM10.5004 12C10.5004 11.5875 10.6473 11.2344 10.941 10.9406C11.2348 10.6469 11.5879 10.5 12.0004 10.5C12.4128 10.5 12.766 10.6469 13.0597 10.9406C13.3535 11.2344 13.5003 11.5875 13.5003 12C13.5003 12.4125 13.3535 12.7656 13.0597 13.0593C12.766 13.3531 12.4128 13.5 12.0004 13.5C11.5879 13.5 11.2348 13.3531 10.941 13.0593C10.6473 12.7656 10.5004 12.4125 10.5004 12ZM16.2696 12C16.2696 11.5875 16.4165 11.2344 16.7102 10.9406C17.004 10.6469 17.3571 10.5 17.7696 10.5C18.1821 10.5 18.5352 10.6469 18.8289 10.9406C19.1227 11.2344 19.2695 11.5875 19.2695 12C19.2695 12.4125 19.1227 12.7656 18.8289 13.0593C18.5352 13.3531 18.1821 13.5 17.7696 13.5C17.3571 13.5 17.004 13.3531 16.7102 13.0593C16.4165 12.7656 16.2696 12.4125 16.2696 12Z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-sjt5nk"
      id="menu-list-:r7:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        class="chakra-menu__menuitem css-1eocwdv"
        data-index="0"
        data-testid="watchlist-rename"
        id="menu-list-:r7:-menuitem-:r8:"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <span
          class="chakra-text css-1rr4qq7"
        >
          Rename watchlist
        </span>
      </button>
      <button
        class="chakra-menu__menuitem css-1eocwdv"
        data-index="1"
        data-testid="watchlist-delete"
        id="menu-list-:r7:-menuitem-:r9:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          class="chakra-text css-rzksxu"
        >
          Delete watchlist
        </span>
        <svg
          class="chakra-icon css-1tqudqa"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.3077 20.4998C6.80898 20.4998 6.38302 20.3232 6.02982 19.97C5.67661 19.6168 5.5 19.1908 5.5 18.6921V5.99981H4.5V4.49983H8.99997V3.61523H15V4.49983H19.5V5.99981H18.5V18.6921C18.5 19.1972 18.325 19.6248 17.975 19.9748C17.625 20.3248 17.1974 20.4998 16.6922 20.4998H7.3077ZM17 5.99981H6.99997V18.6921C6.99997 18.7818 7.02883 18.8556 7.08652 18.9133C7.14422 18.971 7.21795 18.9998 7.3077 18.9998H16.6922C16.7692 18.9998 16.8397 18.9678 16.9038 18.9037C16.9679 18.8395 17 18.769 17 18.6921V5.99981ZM9.40385 16.9998H10.9038V7.99981H9.40385V16.9998ZM13.0961 16.9998H14.5961V7.99981H13.0961V16.9998Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly and click rename 2`] = `
<div>
  <button
    aria-controls="menu-list-:rf:"
    aria-expanded="false"
    aria-haspopup="menu"
    class="chakra-button chakra-menu__menu-button css-1htmeuh"
    id="menu-button-:rf:"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-38f4wb"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M4.73118 12C4.73118 11.5875 4.87805 11.2344 5.17178 10.9406C5.46553 10.6469 5.81865 10.5 6.23113 10.5C6.64363 10.5 6.99675 10.6469 7.29048 10.9406C7.58423 11.2344 7.73111 11.5875 7.73111 12C7.73111 12.4125 7.58423 12.7656 7.29048 13.0593C6.99675 13.3531 6.64363 13.4999 6.23113 13.4999C5.81865 13.4999 5.46553 13.3531 5.17178 13.0593C4.87805 12.7656 4.73118 12.4125 4.73118 12ZM10.5004 12C10.5004 11.5875 10.6473 11.2344 10.941 10.9406C11.2348 10.6469 11.5879 10.5 12.0004 10.5C12.4128 10.5 12.766 10.6469 13.0597 10.9406C13.3535 11.2344 13.5003 11.5875 13.5003 12C13.5003 12.4125 13.3535 12.7656 13.0597 13.0593C12.766 13.3531 12.4128 13.5 12.0004 13.5C11.5879 13.5 11.2348 13.3531 10.941 13.0593C10.6473 12.7656 10.5004 12.4125 10.5004 12ZM16.2696 12C16.2696 11.5875 16.4165 11.2344 16.7102 10.9406C17.004 10.6469 17.3571 10.5 17.7696 10.5C18.1821 10.5 18.5352 10.6469 18.8289 10.9406C19.1227 11.2344 19.2695 11.5875 19.2695 12C19.2695 12.4125 19.1227 12.7656 18.8289 13.0593C18.5352 13.3531 18.1821 13.5 17.7696 13.5C17.3571 13.5 17.004 13.3531 16.7102 13.0593C16.4165 12.7656 16.2696 12.4125 16.2696 12Z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-sjt5nk"
      id="menu-list-:rf:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        class="chakra-menu__menuitem css-1eocwdv"
        data-index="0"
        data-testid="watchlist-rename"
        id="menu-list-:rf:-menuitem-:rg:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          class="chakra-text css-1rr4qq7"
        >
          Rename watchlist
        </span>
      </button>
      <button
        class="chakra-menu__menuitem css-1eocwdv"
        data-index="1"
        data-testid="watchlist-delete"
        id="menu-list-:rf:-menuitem-:rh:"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <span
          class="chakra-text css-rzksxu"
        >
          Delete watchlist
        </span>
        <svg
          class="chakra-icon css-1tqudqa"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.3077 20.4998C6.80898 20.4998 6.38302 20.3232 6.02982 19.97C5.67661 19.6168 5.5 19.1908 5.5 18.6921V5.99981H4.5V4.49983H8.99997V3.61523H15V4.49983H19.5V5.99981H18.5V18.6921C18.5 19.1972 18.325 19.6248 17.975 19.9748C17.625 20.3248 17.1974 20.4998 16.6922 20.4998H7.3077ZM17 5.99981H6.99997V18.6921C6.99997 18.7818 7.02883 18.8556 7.08652 18.9133C7.14422 18.971 7.21795 18.9998 7.3077 18.9998H16.6922C16.7692 18.9998 16.8397 18.9678 16.9038 18.9037C16.9679 18.8395 17 18.769 17 18.6921V5.99981ZM9.40385 16.9998H10.9038V7.99981H9.40385V16.9998ZM13.0961 16.9998H14.5961V7.99981H13.0961V16.9998Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly and click rename 3`] = `
<div>
  <button
    aria-controls="menu-list-:rn:"
    aria-expanded="false"
    aria-haspopup="menu"
    class="chakra-button chakra-menu__menu-button css-1htmeuh"
    id="menu-button-:rn:"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-38f4wb"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M4.73118 12C4.73118 11.5875 4.87805 11.2344 5.17178 10.9406C5.46553 10.6469 5.81865 10.5 6.23113 10.5C6.64363 10.5 6.99675 10.6469 7.29048 10.9406C7.58423 11.2344 7.73111 11.5875 7.73111 12C7.73111 12.4125 7.58423 12.7656 7.29048 13.0593C6.99675 13.3531 6.64363 13.4999 6.23113 13.4999C5.81865 13.4999 5.46553 13.3531 5.17178 13.0593C4.87805 12.7656 4.73118 12.4125 4.73118 12ZM10.5004 12C10.5004 11.5875 10.6473 11.2344 10.941 10.9406C11.2348 10.6469 11.5879 10.5 12.0004 10.5C12.4128 10.5 12.766 10.6469 13.0597 10.9406C13.3535 11.2344 13.5003 11.5875 13.5003 12C13.5003 12.4125 13.3535 12.7656 13.0597 13.0593C12.766 13.3531 12.4128 13.5 12.0004 13.5C11.5879 13.5 11.2348 13.3531 10.941 13.0593C10.6473 12.7656 10.5004 12.4125 10.5004 12ZM16.2696 12C16.2696 11.5875 16.4165 11.2344 16.7102 10.9406C17.004 10.6469 17.3571 10.5 17.7696 10.5C18.1821 10.5 18.5352 10.6469 18.8289 10.9406C19.1227 11.2344 19.2695 11.5875 19.2695 12C19.2695 12.4125 19.1227 12.7656 18.8289 13.0593C18.5352 13.3531 18.1821 13.5 17.7696 13.5C17.3571 13.5 17.004 13.3531 16.7102 13.0593C16.4165 12.7656 16.2696 12.4125 16.2696 12Z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-sjt5nk"
      id="menu-list-:rn:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        class="chakra-menu__menuitem css-1eocwdv"
        data-index="0"
        data-testid="watchlist-rename"
        id="menu-list-:rn:-menuitem-:ro:"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        <span
          class="chakra-text css-1rr4qq7"
        >
          Rename watchlist
        </span>
      </button>
      <button
        class="chakra-menu__menuitem css-1eocwdv"
        data-index="1"
        data-testid="watchlist-delete"
        id="menu-list-:rn:-menuitem-:rp:"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        <span
          class="chakra-text css-rzksxu"
        >
          Delete watchlist
        </span>
        <svg
          class="chakra-icon css-1tqudqa"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.3077 20.4998C6.80898 20.4998 6.38302 20.3232 6.02982 19.97C5.67661 19.6168 5.5 19.1908 5.5 18.6921V5.99981H4.5V4.49983H8.99997V3.61523H15V4.49983H19.5V5.99981H18.5V18.6921C18.5 19.1972 18.325 19.6248 17.975 19.9748C17.625 20.3248 17.1974 20.4998 16.6922 20.4998H7.3077ZM17 5.99981H6.99997V18.6921C6.99997 18.7818 7.02883 18.8556 7.08652 18.9133C7.14422 18.971 7.21795 18.9998 7.3077 18.9998H16.6922C16.7692 18.9998 16.8397 18.9678 16.9038 18.9037C16.9679 18.8395 17 18.769 17 18.6921V5.99981ZM9.40385 16.9998H10.9038V7.99981H9.40385V16.9998ZM13.0961 16.9998H14.5961V7.99981H13.0961V16.9998Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
